doc/developer-guides/interrupt-hld.rst

```diff
@@ -0,0 +1,476 @@
+.. _interrupt-hld:
+
+Interrupt Management high-level design
+######################################
+
+
+Overview
+********
+
+This document describes the interrupt management high-level design for
+the ACRN hypervisor.
+
+The ACRN hypervisor implements a simple but fully functional framework
+to manage interrupts and exceptions, as show in
+:numref:`interrupt-modules-overview`. In its native layer, it configures
+the physical PIC, IOAPIC, and LAPIC to support different interrupt
+sources from local timer/IPI to external INTx/MSI. In its virtual guest
+layer, it emulates virtual PIC, virtual IOAPIC and virtual LAPIC, and
+provides full APIs allowing virtual interrupt injection from emulated or
+pass-thru devices.
+
+.. figure:: images/interrupt-image3.png
+   :align: center
+   :width: 600px
+   :name: interrupt-modules-overview
+
+   ACRN Interrupt Modules Overview
+
+In the software modules view shown in :numref:`interrupt-sw-modules`,
+the ACRN hypervisor sets up the physical interrupt in its basic
+interrupt modules (e.g., IOAPIC/LAPIC/IDT).  It dispatches the interrupt
+in the hypervisor interrupt flow control layer to the corresponding
+handlers, that could be pre-defined IPI notification, timer, or runtime
+registered pass-thru devices.  The ACRN hypervisor then uses its VM
+interfaces based on vPIC, vIOAPIC, and vMSI modules, to inject the
+necessary virtual interrupt into the specific VM
+
+.. figure:: images/interrupt-image2.png
+   :align: center
+   :width: 600px
+   :name: interrupt-sw-modules
+
+   ACRN Interrupt SW Modules Overview
+
+Hypervisor Physical Interrupt Management
+****************************************
+
+The ACRN hypervisor is responsible for all the physical interrupt
+handling. All physical interrupts are first handled in VMX root-mode.
+The "external-interrupt exiting" bit in the VM-Execution controls field
```

Comment:
I'm a little confused by this sentence and wondering if replacing "controls" by "control" would be appropriate/correct?